check-optional-dependencies.js

```diff
@@ -1,14 +1,15 @@
 // check-optional-dependencies.js
-const fs = require('fs');
-const semver = require('semver');
-const myPackageJson = require('./package.json');
+import fs from 'node:fs';
```

Comment:
we should be using [fs/promises](https://nodejs.org/docs/latest-v18.x/api/fs.html#promise-example)